src/sacn/merge_receiver.c

```diff
@@ -604,12 +603,114 @@ etcpal_error_t sacn_merge_receiver_get_source_cid(sacn_merge_receiver_t handle,
 void universe_data(sacn_receiver_t handle, const EtcPalSockAddr* source_addr, const SacnHeaderData* header,
                    const uint8_t* pdata, bool is_sampling, void* context)
 {
-  ETCPAL_UNUSED_ARG(handle);
-  ETCPAL_UNUSED_ARG(source_addr);
-  ETCPAL_UNUSED_ARG(header);
-  ETCPAL_UNUSED_ARG(pdata);
-  ETCPAL_UNUSED_ARG(is_sampling);
   ETCPAL_UNUSED_ARG(context);
+
+  // Look up merger/source info & update pending status if source exists.
+  sacn_source_id_t source_id = SACN_DMX_MERGER_SOURCE_INVALID;
+  sacn_dmx_merger_t merger_handle = SACN_DMX_MERGER_INVALID;
+  bool use_pap = false;
+  if (sacn_lock())
+  {
+    SacnMergeReceiver* merge_receiver = NULL;
+    if (lookup_merge_receiver((sacn_merge_receiver_t)handle, &merge_receiver, NULL) == kEtcPalErrOk)
+    {
+      merger_handle = merge_receiver->merger_handle;
+      use_pap = merge_receiver->use_pap;
+
+      SacnMergeReceiverSource* source = NULL;
+      if (lookup_merge_receiver_source(merge_receiver, &header->cid, &source) == kEtcPalErrOk)
+      {
+        source_id = source->id;
+
+        if (source->pending && (header->start_code == 0x00))
```

Comment:
Let's add a comment here about the receiver always pushing 0x00 after 0xdd or timeout.  Otherwise my first impression was "what about pap"?

Reply:
https://github.com/ETCLabs/sACN/commit/9a85950e3e4cd61be7c675e35a396f3da586b2f5